### PR TITLE
Refactor hit escape moves to use moveend

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Refactors `EFFECT_HIT_ESCAPE` to use `moveend`. This follows the research from smogon but is also generally nice because some additional edge case handling is much nicer now since everything that switches mons out is in `moveend` now.

### CODE REVIEW

File: test/battle/move_effect/hit_escape.c [hunk 1]
```diff
@@ -179,3 +179,18 @@ SINGLE_BATTLE_TEST("Hit Escape: Electric Seed boost is received by the right pok
...
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_ROCKY_HELMET); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_U_TURN); SEND_OUT(player, 1); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_U_TURN, player);
+        HP_BAR(opponent);
+    }
+}
+
```
Comment (Bassoonian) [inline test/battle/move_effect/hit_escape.c:196]:
```suggestion
}
```
